src/world/common/enemy/GrooveGuyAI.inc.c

```diff
@@ -0,0 +1,158 @@
+#include "common.h"
+#include "npc.h"
+
+void N(GrooveGuyAI_02)(Evt* script, NpcAISettings* aiSettings, EnemyTerritoryThing* territory) {
+    Enemy* enemy = script->owner1.enemy;
+    Npc* npc = get_npc_unsafe(enemy->npcID);
+
+    npc->duration = 0;
+    set_npc_yaw(npc, 270.0f);
+    enemy->varTable[0] = 0;
+    script->functionTemp[0] = 3;
+}
+
+void N(GrooveGuyAI_03)(Evt* script, NpcAISettings* aiSettings, EnemyTerritoryThing* territory) {
+    Enemy* enemy = script->owner1.enemy;
+    Npc* npc = get_npc_unsafe((s32) enemy->npcID);
+    s32 phase;
+
+    switch (enemy->varTable[0]) {
+        case 0:
+            enemy->varTable[0] = 1;
+            enemy->varTable[1] = 0;
+            npc->currentAnim.w = 0x3C000C;
+            set_npc_yaw(npc, 270.0f);
+            npc->rotation.y = 0.0f;
+            // fallthrough
+        case 1:
+            phase = enemy->varTable[1] % 16;
+            if (phase < 4) {
+                npc->currentAnim.w = 0x3C000C;
+            }
+            else if (phase < 8) {
```

Comment:
Closing brace on same line as else
```suggestion
            } else if (phase < 8) {
```
& for the other cases